# Data Collection and Inspection

## Contents

[1. Importing Data](#1.-Importing-Data)<br>
[2. Data Inspection](#2.-Data-Inspection) <br>
[3. Data Cleaning](#3.-Data-Cleaning)

### 1. Importing Data

In [1]:
import pandas as pd
pd.set_option('display.max_columns', 9999)
pd.set_option('display.max_rows', 300)

In [3]:
nba_data = pd.read_csv('../data/2012-18_officialBoxScore.csv')

### 2. Data Inspection

In [4]:
nba_data.shape

(44284, 120)

In [5]:
nba_data.head()

gmDate  seasonYR gmTime  seasTyp    offLNm    offFNm teamAbbr teamConf  \
0  10/30/12      2012  19:00  Regular  Brothers      Tony      WAS     East   
1  10/30/12      2012  19:00  Regular     Smith   Michael      WAS     East   
2  10/30/12      2012  19:00  Regular   Workman  Haywoode      WAS     East   
3  10/30/12      2012  19:00  Regular  Brothers      Tony      CLE     East   
4  10/30/12      2012  19:00  Regular     Smith   Michael      CLE     East   

     teamDiv teamLoc teamRslt  teamMin  teamDayOff  teamPTS  teamAST  teamTO  \
0  Southeast    Away     Loss      240           0       84       26      13   
1  Southeast    Away     Loss      240           0       84       26      13   
2  Southeast    Away     Loss      240           0       84       26      13   
3    Central    Home      Win      240           0       94       22      21   
4    Central    Home      Win      240           0       94       22      21   

   teamSTL  teamBLK  teamPF  teamFGA  teamFGM  teamFG%  team2PA  team2PM  \
0       11       10      19       90       32   0.3556       58       24   
1       11       10      19       90       32   0.3556       58       24   
2       11       10      19       90       32   0.3556       58       24   
3        7        5      21       79       36   0.4557       59       29   
4        7        5      21       79       36   0.4557       59       29   

   team2P%  team3PA  team3PM  team3P%  teamFTA  teamFTM  teamFT%  teamORB  \
0   0.4138       32        8     0.25       20       12   0.6000       18   
1   0.4138       32        8     0.25       20       12   0.6000       18   
2   0.4138       32        8     0.25       20       12   0.6000       18   
3   0.4915       20        7     0.35       22       15   0.6818       18   
4   0.4915       20        7     0.35       22       15   0.6818       18   

   teamDRB  teamTRB  teamPTS1  teamPTS2  teamPTS3  teamPTS4  teamPTS5  \
0       21       39        24        15        23        22         0   
1       21       39        24        15        23        22         0   
2       21       39        24        15        23        22         0   
3       36       54        31        19        24        20         0   
4       36       54        31        19        24        20         0   

   teamPTS6  teamPTS7  teamPTS8  teamTREB%  teamASST%  teamTS%  teamEFG%  \
0         0         0         0    41.9355    81.2500   0.4251       0.4   
1         0         0         0    41.9355    81.2500   0.4251       0.4   
2         0         0         0    41.9355    81.2500   0.4251       0.4   
3         0         0         0    58.0645    61.1111   0.5300       0.5   
4         0         0         0    58.0645    61.1111   0.5300       0.5   

   teamOREB%  teamDREB%  teamTO%  teamSTL%  teamBLK%  teamBLKR  teamPPS  \
0    33.3333    53.8462  11.6279   12.3678   11.2434   17.2414   0.9333   
1    33.3333    53.8462  11.6279   12.3678   11.2434   17.2414   0.9333   
2    33.3333    53.8462  11.6279   12.3678   11.2434   17.2414   0.9333   
3    46.1538    66.6667  19.1466    7.8704    5.6217    8.4746   1.1899   
4    46.1538    66.6667  19.1466    7.8704    5.6217    8.4746   1.1899   

   teamFIC  teamFIC40  teamOrtg  teamDrtg  teamEDiff  teamPlay%   teamAR  \
0    67.25    56.0417   94.4447  105.6882   -11.2435     0.3765  18.8679   
1    67.25    56.0417   94.4447  105.6882   -11.2435     0.3765  18.8679   
2    67.25    56.0417   94.4447  105.6882   -11.2435     0.3765  18.8679   
3    74.00    61.6667  105.6882   94.4447    11.2435     0.4390  16.7072   
4    74.00    61.6667  105.6882   94.4447    11.2435     0.4390  16.7072   

   teamAST/TO  teamSTL/TO opptAbbr opptConf    opptDiv opptLoc opptRslt  \
0      2.0000     84.6154      CLE     East    Central    Home      Win   
1      2.0000     84.6154      CLE     East    Central    Home      Win   
2      2.0000     84.6154      CLE     East    Central    Home      Win   
3      1.0476     33.3333

The dataset contains 44,284 rows and 120 columns.  Each game is represented by 6 lines.  This is due to for each game a team is list twice, once under teamAbbr and once under opptAbbr, or once as the winner and once as the loser of the game.  In addition, the three referees are listed separately for each of the games.  The dataset will be condensed two lines for each team.

In [6]:
nba_data['seasonYR'].unique()

array([2012, 2013, 2014, 2015, 2016, 2017])

The data covers the 2012 to 2017 seasons.

In [6]:
nba_data['seasTyp'].unique()

array(['Regular'], dtype=object)

All of the games are for the regular season.

In [7]:
nba_data.isnull().sum()

gmDate        0
seasonYR      0
gmTime        0
seasTyp       0
offLNm        0
offFNm        0
teamAbbr      0
teamConf      0
teamDiv       0
teamLoc       0
teamRslt      0
teamMin       0
teamDayOff    0
teamPTS       0
teamAST       0
teamTO        0
teamSTL       0
teamBLK       0
teamPF        0
teamFGA       0
teamFGM       0
teamFG%       0
team2PA       0
team2PM       0
team2P%       0
team3PA       0
team3PM       0
team3P%       0
teamFTA       0
teamFTM       0
teamFT%       0
teamORB       0
teamDRB       0
teamTRB       0
teamPTS1      0
teamPTS2      0
teamPTS3      0
teamPTS4      0
teamPTS5      0
teamPTS6      0
teamPTS7      0
teamPTS8      0
teamTREB%     0
teamASST%     0
teamTS%       0
teamEFG%      0
teamOREB%     0
teamDREB%     0
teamTO%       0
teamSTL%      0
teamBLK%      0
teamBLKR      0
teamPPS       0
teamFIC       0
teamFIC40     0
teamOrtg      0
teamDrtg      0
teamEDiff     0
teamPlay%     0
teamAR        0
teamAST/TO    0
teamSTL/TO    0
opptAbbr

The data does not contain any nulls.

In [8]:
nba_data.dtypes

gmDate         object
seasonYR        int64
gmTime         object
seasTyp        object
offLNm         object
offFNm         object
teamAbbr       object
teamConf       object
teamDiv        object
teamLoc        object
teamRslt       object
teamMin         int64
teamDayOff      int64
teamPTS         int64
teamAST         int64
teamTO          int64
teamSTL         int64
teamBLK         int64
teamPF          int64
teamFGA         int64
teamFGM         int64
teamFG%       float64
team2PA         int64
team2PM         int64
team2P%       float64
team3PA         int64
team3PM         int64
team3P%       float64
teamFTA         int64
teamFTM         int64
teamFT%       float64
teamORB         int64
teamDRB         int64
teamTRB         int64
teamPTS1        int64
teamPTS2        int64
teamPTS3        int64
teamPTS4        int64
teamPTS5        int64
teamPTS6        int64
teamPTS7        int64
teamPTS8        int64
teamTREB%     float64
teamASST%     float64
teamTS%       float64
teamEFG%  

### 3. Data Cleaning

In [9]:
#Filtering the dataframe to only the fields that will be used.
nba_data = nba_data[['gmDate', 'seasonYR','teamAbbr','teamLoc','teamRslt','teamDayOff',
                    'teamTO','teamPF','teamFGA','teamFGM','teamFTA','teamFTM',
                    'teamORB','teamDRB','teamEFG%','teamTREB%','teamTO%','teamFT%']]

With the referees removed the games can be compressed to two lines for each game by removing duplicates.

In [10]:
nba_data = nba_data.drop_duplicates()
nba_data = nba_data.reset_index(drop=True)
nba_data['gmDate'] = pd.to_datetime(nba_data['gmDate'])
nba_data.head()

gmDate  seasonYR teamAbbr teamLoc teamRslt  teamDayOff  teamTO  teamPF  \
0 2012-10-30      2012      WAS    Away     Loss           0      13      19   
1 2012-10-30      2012      CLE    Home      Win           0      21      21   
2 2012-10-30      2012      BOS    Away     Loss           0      16      23   
3 2012-10-30      2012      MIA    Home      Win           0       8      20   
4 2012-10-30      2012      DAL    Away      Win           0      12      25   

   teamFGA  teamFGM  teamFTA  teamFTM  teamORB  teamDRB  teamEFG%  teamTREB%  \
0       90       32       20       12       18       21    0.4000    41.9355   
1       79       36       22       15       18       36    0.5000    58.0645   
2       75       39       28       23        7       34    0.5600    53.2468   
3       79       43       32       26        5       31    0.5949    46.7532   
4       85       40       18       14        9       31    0.5000    46.5116   

   teamTO%  teamFT%  
0  11.6279   0.6000  
1  19.1466   0.6818  
2  15.4859   0.8214  
3   7.9145   0.8125  
4  11.4373   0.7778

In [11]:
nba_data.shape

(14758, 18)

In [12]:
#Converted the teamRsl from Win and Loss to O for Loss and 1 for Win.
#Converted the teamLoc from Home and Away to O for Away and 1 for Home.
nba_data['teamRslt'] = nba_data['teamRslt'].apply(lambda x: 0 if x=='Loss' else 1)
nba_data['teamLoc'] = nba_data['teamLoc'].apply(lambda x: 0 if x=='Away' else 1)

In [13]:
#Coverting two of the percentage columns from decimals
nba_data['teamEFG%'] = round((nba_data['teamEFG%'] * 100),4)
nba_data['teamFT%'] = round((nba_data['teamFT%'] * 100),4)

In [14]:
#Saving data to a new file
nba_data.to_csv('./data/nba_data.csv', index=False)